packages/amplify-category-geo/src/service-utils/prepare-app.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * This function is taken from : https://github.com/aws/aws-cdk/blob/cea1039e3664fdfa89c6f00cdaeb1a0185a12678/packages/%40aws-cdk/core/lib/private/prepare-app.ts#L17
+ * In cdkV2 this function is not exported by the library. This function is copied from above to get previous behavior
+ */
+
+import { IConstruct, Dependable } from 'constructs';
+import { CfnResource } from 'aws-cdk-lib';
+
+/**
+ * Prepares the app for synthesis. This function is called by the root `prepare`
+ * (normally this the App, but if a Stack is a root, it is called by the stack),
+ * which means it's the last 'prepare' that executes.
+ *
+ * It takes care of reunifying cross-references between stacks (or nested stacks),
+ * and of creating assets for nested stack templates.
+ */
+export const prepareApp = (root: IConstruct): void => {
```

Comment:
I suggest to:
- rename this from `prepareApp` to `cdkPrepareAppShim` or even `cdkv1PrepareAppShim` to discourage usage
- consider marking it deprecated and point people towards cdk.App as alternative
- I'd rename file from `prepare-app.ts` to `cdkv1-compat.ts`